node/bindnode/node.go

```diff
@@ -571,15 +643,19 @@ func (w *_assembler) AssignInt(i int64) error {
 }
 
 func (w *_assembler) AssignFloat(f float64) error {
-	if w.kind() != datamodel.Kind_Float {
-		return datamodel.ErrWrongKind{
-			TypeName:        w.schemaType.Name(),
-			MethodName:      "AssignFloat",
-			AppropriateKind: datamodel.KindSet{datamodel.Kind_Float},
-			ActualKind:      w.kind(),
+	if err := compatibleKind(w.schemaType, datamodel.Kind_Float); err != nil {
+		return err
+	}
+	if _, ok := w.schemaType.(*schema.TypeAny); ok {
+		basicBuilder := basicnode.Prototype.Any.NewBuilder()
+		if err := basicBuilder.AssignFloat(f); err != nil {
+			return err
 		}
+		basicNode := basicBuilder.Build()
```

Comment:
You could use `basicnode.NewFloat` in this and similar methods for some slight improvement to terseness.  (Also it will do less allocating because it skips the builder.)

Reply:
Excellent callout, thank you.